tests/test_volumeconstraint.py

```diff
@@ -0,0 +1,126 @@
+import pytest
+import firedrake as fd
+import fireshape as fs
+import fireshape.zoo as fsz
+import ROL
+
+
+@pytest.mark.parametrize("inner_t", [fs.H1InnerProduct, fs.ElasticityInnerProduct, fs.LaplaceInnerProduct])
+@pytest.mark.parametrize("use_extension", ["wo_ext", "w_ext", "w_ext_fixed_fim"])
+@pytest.mark.parametrize("controlspace_t", [fs.FeControlSpace, fs.FeMultiGridControlSpace, fs.BsplineControlSpace])
+@pytest.mark.parametrize("dim", [2, 3])
+def test_levelset(dim, inner_t, controlspace_t, use_extension, pytestconfig):
+    verbose = pytestconfig.getoption("verbose")
+    """ Test template for fsz.LevelsetFunctional."""
+
+    clscale = 0.1 if dim == 2 else 0.2
+
+    # make the mesh a bit coarser if we are using a multigrid control space as
+    # we are refining anyway
+    if controlspace_t == fs.FeMultiGridControlSpace:
+        clscale *= 4
+
+    if dim == 2:
+        mesh = fs.DiskMesh(0.1)
+    elif dim == 3:
+        mesh = fs.SphereMesh(0.2)
+    else:
+        raise NotImplementedError
+
```

Comment:
`clscale` is never used. I guess that's because this part was just copied over from `test_levelset.py`, where it is also never used. we should fix that (i.e. 
```
    if dim == 2:
        mesh = fs.DiskMesh(clscale)
    elif dim == 3:
        mesh = fs.SphereMesh(clscale)
```

Reply:
I'll do it now